lib/ClangImporter/ImportName.cpp

```diff
@@ -1600,6 +1610,15 @@ ImportedName NameImporter::importNameImpl(const clang::NamedDecl *D,
   ArrayRef<const clang::ParmVarDecl *> params;
   switch (D->getDeclName().getNameKind()) {
   case clang::DeclarationName::CXXConstructorName:
+    isInitializer = true;
+    isFunction = true;
+    result.info.initKind = CtorInitializerKind::Designated;
+    baseName = "init";
+    if (auto ctor = dyn_cast<clang::CXXConstructorDecl>(D)) {
```

Comment:
I don't think this cast can fail for declarations with a CXXConstructorName name.